### PR TITLE
Kelly schroeder 1801 final

### PR DESCRIPTION
**Pull request by anubiskhan**

### CODE REVIEW

**Comment by anubiskhan** [comment]
I think I did very well on today's assessment. I did miss the final line of the fourth user story though. I was unclear on whether or not genre creation functionality needed to be nested under admin, but the user story stated as an unregistered user ```I cannot see the form to create new genres, ``` it didn't state that an an unregistered user is unable to create a new genre. Because of this I believe the branch logic hiding the form completes the user story satisfactorily

File: app/controllers/genres_controller.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  def show
+    @genre = Genre.find(params[:id])
+  end
+
+  def create
+    @genre = Genre.new(genre_params)
+    if @genre.save
+      redirect_to genres_path
+    else
+      redirect_to genres_path
```
Comment (AliSchlereth) [inline app/controllers/genres_controller.rb]: How would a user know whether their genre was created or not? Both the happy and sad paths do the same thing here.

Comment (AliSchlereth) [reply app/controllers/genres_controller.rb]: I see your reasoning for not scoping the create action to an admin, but we'd like you to implement that piece of the functionality none the less.

File: app/views/genres/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<main>
+  <% if current_admin? %>
+    <%= render partial: "form" %>
+  <% end %>
+<h2>Genres</h2>
+  <% @genres.each do |g| %>
+    <a href="/genres/<%= g.id %>"><%= g.name %></a>
```
Comment (AliSchlereth) [inline app/views/genres/index.html.erb]: Why didn't you use a link helper here?

**Comment by AliSchlereth** [comment]
@anubiskhan This is pretty good, but we have a few things we'd like you to update:
* Only an admin should be able to create a genre. Right now if someone sent a POST to /genres they'd be able to make a genre even if not an admin.
* How would you let a user know if their genre wasn't created successfully? Make sure to add a sad path test for this functionality too.
* Please finish the last user story.

**Pull request closed without merging**